src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -1320,8 +1388,51 @@ std::optional<size_t> ArrowParquetSchemaReader::readNumberOrRows()
     return metadata->num_rows();
 }
 
+std::vector<FileBucketInfoPtr> ParquetBucketSplitter::splitToBuckets(size_t bucket_size, ReadBuffer & buf, const FormatSettings & format_settings_)
+{
+    std::atomic<int> is_stopped = false;
+    auto arrow_file = asArrowFile(buf, format_settings_, is_stopped, "Parquet", PARQUET_MAGIC_BYTES, /* avoid_buffering */ true, nullptr);
+    auto metadata = parquet::ReadMetaData(arrow_file);
+    std::vector<size_t> bucket_sizes;
+    for (int i = 0; i < metadata->num_row_groups(); ++i)
+        bucket_sizes.push_back(metadata->RowGroup(i)->total_byte_size());
+
+    std::vector<std::vector<size_t>> buckets;
+    size_t current_weight = 0;
+    for (size_t i = 0; i < bucket_sizes.size(); ++i)
+    {
+        if (current_weight + bucket_sizes[i] <= bucket_size)
+        {
+            buckets.back().push_back(i);
```

Comment:
Now here if we are on the first iteration, we will call `back()` on an empty vector
```suggestion
            if (buckets.empty())
                buckets.emplace_back();
            buckets.back().push_back(i);
```

Reply:
my bad...